File: mvvm-java/common/src/main/java/com/wasisto/githubuserfinder/common/Logger.java
package com.wasisto.githubuserfinder.common;

public interface Logger {

    void trace(String message);

    void trace(String message, Throwable throwable);

    void debug(String message);

    void debug(String message, Throwable throwable);

    void info(String message);

    void info(String message, Throwable throwable);

    void warn(String message);

    void warn(String message, Throwable throwable);

    void error(String message);

    void error(String message, Throwable throwable);

    void fatal(String message);

    void fatal(String message, Throwable throwable);
}

File: mvp-kotlin/common/src/main/java/com/wasisto/githubuserfinder/common/Logger.kt
package com.wasisto.githubuserfinder.common

interface Logger {

    fun trace(message: String)

    fun trace(message: String, throwable: Throwable)

    fun debug(message: String)

    fun debug(message: String, throwable: Throwable)

    fun info(message: String)

    fun info(message: String, throwable: Throwable)

    fun warn(message: String)

    fun warn(message: String, throwable: Throwable)

    fun error(message: String)

    fun error(message: String, throwable: Throwable)

    fun fatal(message: String)

    fun fatal(message: String, throwable: Throwable)
}